users/zanna/fake_currency_ML_zanna.py

```diff
@@ -0,0 +1,163 @@
+# Dataset from kaggle.com Fake Currency Data. Contains synthetic data representing fake currency samples. Classification
+
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pylab as plt
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import cross_val_score
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+
+df = pd.read_csv('users/zanna/fake_currency_data.csv')
+
+#Colums:
+#Country: Country of origin for the currency.
+#Denomination: Currency denomination.
+#Counterfeit: Binary indicator (0 for genuine, 1 for counterfeit).
+#SerialNumber: Serial number of the currency.
+#Security Features: Security features present in the currency.
+#Weight: Weight of the currency in grams.
+#Length: Length of the currency in mm.
+#Width: Width of the currency in mm.
+#Thickness: Thickness of the currency in mm
+
+df.dtypes #checking the data types, Country, Denomination, SecurityFeatures are object, rest is int/float
+df[['Country','Denomination','SecurityFeatures']] #checking if these 3 features should be 'object', Denomination could be split into value and the currency
+df.describe() #counterfeit min:0,max:1, mean:0.499, median:0, Q75:1 -> there are more genuine banknotes, but the difference in quantity small
+df['Counterfeit'].value_counts()
+df['Country'].value_counts()
+#as the general describe() didn't show all columns, the range had to be narrowed
+df[['Weight','Length','Width','Thickness']].describe() #judging by the statistics there are no outliers
+
+df.Counterfeit.unique() #Counterfeit contains only 0 and 1
+df.Country.unique() #USA, EU, UK -> one-hot encoding, then the currency wont be needed
+df.SecurityFeatures.unique() #Hologram, Security Thread, Microprint, Watermark ->one-hot encoding
+df.isnull().sum() #there are no nulls
+
+df['D'] = df['Denomination'].str[1:]
+
+#one-hot encoding for Country and SecurityFeatures
+df_dummies=pd.get_dummies(df[['Country','SecurityFeatures']])
+df_encoded = pd.concat([df, df_dummies], axis=1)
+df_encoded.dtypes
+df_encoded = df_encoded.drop(['Country', 'SecurityFeatures','Denomination','SerialNumber'], axis=1)
+df_encoded.D = df_encoded.D.str.replace(',', '').astype(int) #converting coulmn D (denomination) into int
+
+corr = df_encoded.corr()
+sns.heatmap(corr, annot=True)
+#correlation is small
+
+#Setting the features and target variable in the dataset
+X = df_encoded.drop('Counterfeit', axis=1)
+y = df_encoded['Counterfeit']
+
+#Creating training and test sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+#Features standarization
+scaler = StandardScaler()
+X_train_scaled = scaler.fit_transform(X_train)
+X_test_scaled = scaler.fit_transform(X_test)
```

Comment:
w przypadku zbioru testowego uzywamy tylko fit, best fit_transform. 

Chodzi o to zeby nasz scaler wytrenowal sie na zbiorze treningowym, a nastepnie na zbiorze testowym uzywamy juz wytrenowanego scalera